packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,820 @@
+          distributionIncentive,
+        },
+      ],
+      transactionOverrides,
+    })
+  }
+
+  protected async _distribute({
+    split,
+    recipients,
+    allocations,
+    distributionIncentive,
+    token,
+    distributor,
+    transactionOverrides = {},
+  }: DistributeSplitConfig): Promise<TransactionFormat> {
+    validateAddress(split)
+    validateAddress(token)
+    validateAddress(distributor)
+    recipients.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const totalAllocation = allocations.reduce((a, b) => a + b)
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2FactoryABI,
+      functionName: 'distribute',
+      functionArgs: [
+        {
+          recipients,
+          allocations,
+          totalAllocation,
+          distributionIncentive,
+        },
+        token,
+        distributor,
+      ],
+      transactionOverrides,
+    })
+  }
+
+  protected _getSplitV2Contract(
+    split: Address,
+  ): GetContractReturnType<SplitV2ABI, PublicClient<Transport, Chain>> {
+    return getContract({
+      address: split,
+      abi: splitV2ABI,
+      publicClient: this._publicClient,
+      walletClient: this._walletClient,
+    })
+  }
+
+  protected _getSplitV2FactoryContract(
+    splitType: SplitV2Type,
+  ): GetContractReturnType<SplitFactoryABI, PublicClient<Transport, Chain>> {
+    return getContract({
+      address: getSplitV2FactoryAddress(this._chainId, splitType),
+      abi: splitV2FactoryABI,
+      publicClient: this._publicClient,
+      walletClient: this._walletClient,
+    })
+  }
+
+  protected async _eip712Domain(
+    split: Address,
+  ): Promise<{ domain: TypedDataDomain }> {
+    this._requirePublicClient()
+
+    const eip712Domain =
+      await this._getSplitV2Contract(split).read.eip712Domain()
+
+    return {
+      domain: {
+        chainId: Number(eip712Domain[3].toString()),
+        name: eip712Domain[1],
+        version: eip712Domain[2],
+        verifyingContract: eip712Domain[4],
+        salt: eip712Domain[5],
+      },
+    }
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class SplitV2Client extends SplitV2Transactions {
+  readonly eventTopics: { [key: string]: Hex[] }
+  readonly callData: SplitV2CallData
+  readonly estimateGas: SplitV2GasEstimates
+  readonly sign: SplitV2Signature
+
+  constructor({
+    chainId,
+    publicClient,
+    ensPublicClient,
+    walletClient,
+    includeEnsNames = false,
+  }: SplitsClientConfig) {
+    super({
+      transactionType: TransactionType.Transaction,
+      chainId,
+      publicClient,
+      ensPublicClient,
+      walletClient,
+      includeEnsNames,
+    })
+
+    if (!SPLITS_SUPPORTED_CHAIN_IDS.includes(chainId))
+      throw new UnsupportedChainIdError(chainId, SPLITS_SUPPORTED_CHAIN_IDS)
```

Comment:
Just remember to update this list of ids based with wherever we deploy.

Reply:
wait, shouldn't `SPLITS_SUPPORTED_CHAIN_IDS` be `SPLITS_V2_SUPPORTED_CHAIN_IDS`?